Dialog/RoomConfigDialog.cpp

```diff
@@ -486,3 +394,129 @@ void RoomConfigDialog::on_spinBox_Layer0Height_valueChanged(int arg1)
         ui->spinBox_Layer0Height->setStyleSheet("");
     }
 }
+
+/// <summary>
+/// Slot function for spinBox_BGLayerScrollingFlag_valueChanged.
+/// </summary>
+/// <param name="arg1">
+/// The spinbox value.
+/// </param>
+void RoomConfigDialog::on_spinBox_BGLayerScrollingFlag_valueChanged(int arg1)
+{
+    switch(arg1)
+    {
+    // TODO: Add more info for other cases, 1-7 are all meaningful cases
+    case 1: ui->label_CurBGLayerScrollingType->setText("No scrolling"); break;
+    case 3: ui->label_CurBGLayerScrollingType->setText("Hide BG Layer"); break;
+    case 7: ui->label_CurBGLayerScrollingType->setText("full BG Layer autoscroll"); break;
+    default: ui->label_CurBGLayerScrollingType->setText("Unknown");
+    }
+}
+
+/// <summary>
+/// Slot function for spinBox_Layer0MappingType_valueChanged.
+/// </summary>
+/// <param name="arg1">
+/// The spinbox value.
+/// </param>
+void RoomConfigDialog::on_spinBox_Layer0MappingType_valueChanged(int arg1)
+{
+    switch(arg1)
+    {
+    case 0x00:
+    case 0x01:
+    case 0x02:
+    case 0x03:
+    case 0x04:
+    case 0x05:
+    case 0x06:
+    case 0x07:
+    case 0x08:
+    case 0x09:
+    case 0x0A:
+    case 0x0B:
+    case 0x0C:
+    case 0x0D:
+    case 0x0E:
+    case 0x0F:
+    {
+        ui->label_CurLayer0MappingType->setText("Disabled"); break;
+    }
+    case 0x10:
+    case 0x13:
+    case 0x14:
+    case 0x15:
+    case 0x16:
+    case 0x17:
+    case 0x18:
+    case 0x19:
+    case 0x1A:
+    case 0x1B:
+    case 0x1C:
+    case 0x1D:
+    case 0x1E:
+    case 0x1F:
+    {
+        ui->spinBox_Layer0Width->setValue(ui->SpinBox_RoomWidth->value());
+        ui->spinBox_Layer0Height->setValue(ui->SpinBox_RoomHeight->value());
+        ui->label_CurLayer0MappingType->setText("Map16");
+        break;
+    }
+    case 0x11: ui->label_CurLayer0MappingType->setText("Map16 & The Big Board result Bar control"); break;
+    case 0x12: ui->label_CurLayer0MappingType->setText("Map16 & asyn cam-based H-scroll"); break;
+    case 0x20:
+    case 0x21:
+    case 0x23:
+    case 0x24:
+    case 0x25:
+    case 0x26:
+    case 0x27:
+    case 0x28:
+    case 0x29:
+    case 0x2A:
+    case 0x2B:
+    case 0x2C:
+    case 0x2D:
+    case 0x2E:
+    case 0x2F:
+    {
+        ui->label_CurLayer0MappingType->setText("Tile8x8"); break;
+    }
+    case 0x22: ui->label_CurLayer0MappingType->setText("Tile8x8 & autoscroll"); break;
+    }
```

Comment:
Let's add a default which prints a warning for unsupported value

Reply:
oh, you are right, but don't need to do this, let me just cap the spinbox's value to be under 0x30.